File: back-end/src/main/java/com/hsuliz/backend/main/expence/ExpenseService.java
package com.hsuliz.backend.main.expence;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class ExpenseService {

    private final ExpenseRepository expenseRepository;


    public Expense getExpense(long id) {
        return expenseRepository
                .findById(id)
                .orElseThrow(RuntimeException::new);
    }

    public List<Expense> getExpenses() {
        return expenseRepository.findAll();
    }

    public void saveExpense(Expense expense) {
        expenseRepository.save(expense);
    }

    public void deleteExpense(long id) {
        expenseRepository.deleteById(id);
    }

}

File: front-end/src/services/ExpenseService.ts
import requester from "./requester";
import {Expense} from "./Types";

const getExpense = (id: number) => {
    return requester.get<Expense>(`/expenses/${id}`)
}

const getExpenses = () => {
    return requester.get<Array<Expense>>("/expenses")
}

const post = (expense: Expense) => {
    return requester.post<Expense>("/expenses", expense)
}

const ExpenseService = {
    getExpense, getExpenses, post
}

export default ExpenseService;